backends/vulkan/test/op_tests/dequantize_test.cpp

```diff
@@ -0,0 +1,316 @@
+    int64_t zero_point,
+    int64_t quant_min,
+    int64_t quant_max,
+    at::ScalarType dtype,
+    at::ScalarType out_dtype) {
+  auto out = at::empty_like(input, out_dtype);
+  // Convert at::ScalarType to executorch::ScalarType
+  ScalarType et_dtype;
+  ScalarType et_out_dtype;
+
+  switch (dtype) {
+    case at::kByte:
+      et_dtype = ScalarType::Byte;
+      break;
+    case at::kChar:
+      et_dtype = ScalarType::Char;
+      break;
+    case at::kShort:
+      et_dtype = ScalarType::Short;
+      break;
+    case at::kInt:
+      et_dtype = ScalarType::Int;
+      break;
+    case at::kLong:
+      et_dtype = ScalarType::Long;
+      break;
+    default:
+      throw std::runtime_error("Unsupported dtype");
+  }
+
+  switch (out_dtype) {
+    case at::kHalf:
+      et_out_dtype = ScalarType::Half;
+      break;
+    case at::kFloat:
+      et_out_dtype = ScalarType::Float;
+      break;
+    case at::kDouble:
+      et_out_dtype = ScalarType::Double;
+      break;
+    default:
+      throw std::runtime_error("Unsupported out_dtype");
+  }
+
+  executorch::aten::optional<ScalarType> opt_et_out_dtype(et_out_dtype);
+
+  WRAP_TO_ATEN(dequantize_per_tensor_out_no_context, 7)
+  (input,
+   scale,
+   zero_point,
+   quant_min,
+   quant_max,
+   et_dtype,
+   opt_et_out_dtype,
+   out);
+  return out;
+}
+
+// ATen wrapper for dequantize_per_token
+at::Tensor dequantize_per_token_aten(
+    const at::Tensor& input,
+    const at::Tensor& scale,
+    const at::Tensor& zero_points,
+    int64_t quant_min,
+    int64_t quant_max,
+    at::ScalarType dtype,
+    at::ScalarType out_dtype) {
+  auto out = at::empty_like(input, out_dtype);
+  // Convert at::ScalarType to executorch::ScalarType
+  ScalarType et_dtype;
+  ScalarType et_out_dtype;
+
+  switch (dtype) {
+    case at::kByte:
+      et_dtype = ScalarType::Byte;
+      break;
+    case at::kChar:
+      et_dtype = ScalarType::Char;
+      break;
+    case at::kShort:
+      et_dtype = ScalarType::Short;
+      break;
+    case at::kInt:
+      et_dtype = ScalarType::Int;
+      break;
+    case at::kLong:
+      et_dtype = ScalarType::Long;
+      break;
+    default:
+      throw std::runtime_error("Unsupported dtype");
+  }
+
+  switch (out_dtype) {
+    case at::kHalf:
+      et_out_dtype = ScalarType::Half;
+      break;
+    case at::kFloat:
+      et_out_dtype = ScalarType::Float;
+      break;
+    case at::kDouble:
+      et_out_dtype = ScalarType::Double;
+      break;
+    default:
+      throw std::runtime_error("Unsupported out_dtype");
+  }
+
+  WRAP_TO_ATEN(dequantize_per_token_out_no_context, 7)
+  (input,
+   scale,
+   zero_points,
+   quant_min,
+   quant_max,
+   et_dtype,
+   et_out_dtype,
+   out);
+  return out;
+}
+
+} // namespace native
+} // namespace executor
+} // namespace torch
+
+//
+// Test functions
+//
+
+// Helper function to get the name of a ScalarType for better error messages
+std::string scalar_type_name(c10::ScalarType dtype) {
```

Comment:
isn't this the same helper function you used for the quantized tests? if so, please refactor this into some utilities.